src/System.Private.ServiceModel/tests/Common/Scenarios/ScenarioTestTypes.cs

```diff
@@ -793,6 +797,30 @@ public Guid CallbackGuid
         }
     }
 
+    public XmlCompositeType XmlCallbackGuid
+    {
+        get
+        {
+            if (_xml_tcs.Task.Wait(ScenarioTestHelpers.TestTimeout))
+            {
+                return _xml_tcs.Task.Result;
+            }
+            throw new TimeoutException(string.Format("Not completed within the alloted time of {0}", ScenarioTestHelpers.TestTimeout));
+        }
+    }
+
+    public ComplexCompositeType DataContractCallbackGuid
+    {
```

Comment:
Done, I also created unique versions as I was re-using these types. I agree that having unique/use once types is better/safer than trying to figure out how they are safe to use by other tests without messing up these tests.